Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationResultDispatchingValidator.cs

```diff
@@ -36,16 +36,71 @@ public void DispatchValidationFailures (IBusinessObjectValidationResult validati
       ArgumentUtility.CheckNotNull("validationResult", validationResult);
 
       var bocListControl = GetControlToValidate();
+      var dispatchToValidationFailureRepository = bocListControl
+          .GetColumnDefinitions()
+          .Any(cd => cd is BocValidationErrorIndicatorColumnDefinition);
 
+      if (dispatchToValidationFailureRepository)
+        DispatchToValidationFailureRepository(validationResult, bocListControl);
+      else
+        DispatchToValidationResultDispatchers(validationResult, bocListControl);
+    }
+
+    private void DispatchToValidationResultDispatchers (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
       var validatorsMatchingToControls = EnumerableUtility.SelectRecursiveDepthFirst(
               bocListControl as Control,
-              child => child.Controls.Cast<Control>().Where(item => !(item is INamingContainer)))
+              child => child.Controls.Cast<Control>().Where(item => item is not INamingContainer))
           .OfType<IBusinessObjectBoundEditableWebControlValidationResultDispatcher>();
 
       foreach (var validator in validatorsMatchingToControls)
         validator.DispatchValidationFailures(validationResult);
     }
 
+    private void DispatchToValidationFailureRepository (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
+      var validationFailureRepository = bocListControl.ValidationFailureRepository;
+      validationFailureRepository.ClearAllValidationFailures();
+
+      var rowObjects = bocListControl.Value?.Cast<IBusinessObject>();
+
+      if (rowObjects != null)
+      {
+        var columnsAndMatchers = bocListControl
+            .GetColumnDefinitions()
+            .OfType<IBocColumnDefinitionWithValidationSupport>()
+            .Select(c => new { Column = (BocColumnDefinition)c, Matcher = c.GetValidationFailureMatcher() })
```

Comment:
These days  (modern C# etc), should we use ValueTuple isntead of Tuple in cases such as this to optimize allocatons a bit?

Reply:
Using a value tuple in a LINQ method is a bit annoying as it cannot detect the type correctly and the naming stuff won't work out of the box. In this case I would keep the anonymous type, these short-lived 15 allocations are nothing against other stuff.